linkerd/trace-context/src/propagation.rs

```diff
@@ -235,3 +341,21 @@ fn try_split_to(buf: &mut Bytes, n: usize) -> Result<Bytes, InsufficientBytes> {
         Err(InsufficientBytes)
     }
 }
+
+#[test]
+fn test_w3c_context_parsed_successfully() {
```

Comment:
should we test the behavior of invalid headers?

Reply:
I've added tests. We're mainly testing the parsing logic (i.e not testing anything that has to do with fetching the actual header value). I've added one or more tests for the obvious failure conditions:

1) Version is invalid (either not 00, or completely missing)
2) Invalid hex for trace/parent id (parsing should fail)
3) When flags are invalid, they get set to `0`.

Think this should be all.